aws/resource_aws_servicecatalog_product_test.go

```diff
@@ -0,0 +1,241 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/servicecatalog"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+func TestAccAWSServiceCatalogProduct_basic(t *testing.T) {
+	arbitraryProductName := fmt.Sprintf("product-%s", acctest.RandString(5))
+	arbitraryProvisionArtifactName := fmt.Sprintf("pa-%s", acctest.RandString(5))
+	arbitraryBucketName := fmt.Sprintf("bucket-%s", acctest.RandString(16))
+	tag1 := "FooKey = \"bar\""
+	tag2 := "BarKey = \"foo\""
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckServiceCatalogProductDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckAwsServiceCatalogProductResourceConfigTemplate(arbitraryBucketName, arbitraryProductName, arbitraryProvisionArtifactName, tag1, tag2),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("aws_servicecatalog_product.test", "name", arbitraryProductName),
```

Comment:
small nit, can you extract this to a variable and define it once?
`resourceName := "aws_servicecatalog_product.test"` and pass `resourceName` to to other funcs

Reply:
done (check-in to come)